src/Packagist/Api/Client.php

```diff
@@ -192,6 +206,34 @@ protected function respond(string $url)
         return $this->create($response);
     }
 
+    /**
+     * Execute two urls request, parse and merge the responses
+     *
+     * @param string $url1
+     * @param string $url2
+     * @return array|Package
+     */
+    protected function multiRespond(string $url1, string $url2)
+    {
+        $response1 = $this->request($url1);
+        $response1 = $this->parse((string)$response1);
+
+        $response2 = $this->request($url2);
+        $response2 = $this->parse((string)$response2);
+
+        foreach ($response1['packages'] as $k1 => $package1) {
+            foreach ($response2['packages'] as $k2 => $package2) {
+                if ($k1 === $k2) {
+                    foreach ($package2 as $version) {
+                        $response1['packages'][$k1][] = $version;
+                    }
+                }
+            }
+        }
```

Comment:
I think there is probably a performance optimisation to be made here, to reduce the level of loop nesting

Reply:
I'll take a look at this today and see if I can make a suggestion. Thanks again for the PR, and sorry again for taking so long to get back to you.

Reply:
@JellyBellyDev from what I can tell the `$package2` we receive in this loop has a different data structure depending on the version, i.e. the first time it contains `name`, `description`, etc, then the following times it doesn't - only containing `version` etc. Did you come across that in your testing?

Reply:
Yep, the data structure it's very strange and I have tried to manage all the cases...